scraping/scraping/spiders/amazon.py

```diff
@@ -0,0 +1,70 @@
+from logging import getLogger
+from typing import Iterator
+
+from scrapy_splash import SplashJsonResponse, SplashRequest
+
+from ..splash import scroll_end_of_page_script
+from ._base import BaseSpider
+
+logger = getLogger(__name__)
+
+
+class AmazonSpider(BaseSpider):
+    name = "amazon"
+    allowed_domains = ["amazon.de"]
+
+    def parse_SERP(self, response: SplashJsonResponse) -> Iterator[SplashRequest]:
+        """
+        The `Scrapy` framework executes this method.
+
+        Yields:
+            Iterator[SplashRequest]: Requests that will be performed to scrap each product page
+        """
+        # Save HTML to database
+        self._save_SERP(response)
+        urls = response.css("div.a-row.a-size-base.a-color-base a::attr(href)").getall()
+        prices = response.css(
+            "div.a-row.a-size-base.a-color-base span.a-price-whole::text"
+        ).getall()
+
+        # Yield request for each product
+        logger.info(f"Number of products to be scraped {len(urls)}")
+
+        for url, price in zip(urls, prices):
+            if "refinements=p_n_cpf_eligible" in url:
+                yield SplashRequest(
+                    url=f"https://www.amazon.de{url}",
+                    callback=self.parse_PRODUCT,
+                    meta={
+                        "amazon_price": price
+                    },  # customized meta_information working used just in amazon
```

Comment:
...and here: `meta={request_meta_information:{"price":price}}` or you can use "amazon_price" as key.

Reply:
Changed in [74f677d](https://github.com/calgo-lab/green-db/pull/65/commits/74f677d372ac18391ade43d100136d7839f61e54)